hudi-integ-test/src/main/java/org/apache/hudi/integ/testsuite/dag/nodes/ValidateDatasetNode.java

```diff
@@ -51,7 +51,7 @@ public Dataset<Row> getDatasetToValidate(SparkSession session, ExecutionContext
                                            StructType inputSchema) {
     String partitionPathField = context.getWriterContext().getProps().getString(DataSourceWriteOptions.PARTITIONPATH_FIELD().key());
     String hudiPath = context.getHoodieTestSuiteWriter().getCfg().targetBasePath + (partitionPathField.isEmpty() ? "/" : "/*/*/*");
-    Dataset<Row> hudiDf = session.read().option(HoodieMetadataConfig.ENABLE.key(), String.valueOf(config.isEnableMetadataValidate()))
+    Dataset<Row> hudiDf = session.read().option(HoodieMetadataConfig.ENABLE.key(), String.valueOf(context.getHoodieTestSuiteWriter().getCfg().enableMetadataOnRead))
```

Comment:
so we prefer `context.getHoodieTestSuiteWriter().getCfg()` over `this.config` ? how would people know this is the preferred way: we either make `this.config` usable or remove it so people only use context to retrieve config?

Reply:
there is some diff here. this.config refers to per node config. where as context.getHoodieTestSuiteWriter().getCfg() refers to HoodieTestSuiteJob.Config. they are not the same.

Reply:
ok so should we make job config take precedence and fall back to node config if not present? looks like some merge logic needed here

Reply:
Here is how I am deciding where to fit the config. 
some of them are just a top level config. like table base path, index type, isMetadata enabled since we wish to apply it to all nodes. while some are node level configs. for eg, num_records_to_insert, "query to use to validate" etc.

Reply:
@nsivabalan Ok took a closer look at the code. The node-level config is `DeltaConfig.Config`, which is a confusing class name. By using `Config` alone in the code makes it hard to tell what it contains. It's actually a bag of configs for the test suite, which should be called something like test suite config.

As for `getHoodieTestSuiteWriter().getCfg()` returning `HoodieTestSuiteJob.HoodieTestSuiteConfig` makes it worse; it's job-level parameters passed to deltastreamer. We should call it `HoodieTestSuiteJobParams` to distinguish from a config object, which is typically a `HoodieConfig` with builder pattern throughout the codebase.

This is not nitpicking; test classes have a lot to do with config pumping logic so we need to be careful designing good APIs and boundaries. We should make some code improvements in a separate pr